src/end.c

```diff
@@ -1553,16 +1553,19 @@ really_done(int how)
              * any vitality to stop it) */
             corpse = mk_named_object(CORPSE, &mons[mnum], u.ux, u.uy, g.plname);
         }
-        if (yn("Do you want to write your own epitaph?") != 'y') {
+        pbuf[0] = '\0';
+        if (yn("Do you want to write your own epitaph?") == 'y') {
+            char ebuf[BUFSZ];
+            getlin("Enter your epitaph:", ebuf);
+            if (*ebuf) {
+                Sprintf(pbuf, "%s.  ", g.plname);
+                (void) strncat(pbuf, ebuf, sizeof pbuf - Strlen(pbuf) - 1);
+            }
+        }
+        if (!*pbuf) { /* didn't write own epitaph, or entered a blank line */
```

Comment:
Okay, but this shouldn't drop the 100 character maximum - the epitaph may be read by someone in a subsequent game and it being >100 characters is inconsistent with other epitaphs.

Reply:
But there isn't a way to enforce a 100 character limit when entering the text, so the only option to do it is to save only a partial/cut-off epitaph (potentially, in the event someone enters more than 100 characters).  I think saving half an epitaph is going to be more broken and noticeable than a changed/inconsistent character limit.

Reply:
Why not enforce it by saying "Your epitaph is too long, please try again with 100 characters or less" or similar?